### PR TITLE
Add proto for `DataColumnIdentifier`, `DataColumnSidecar`, `DataColumnSidecarsByRangeRequest` and `MetadataV2`.

### PR DESCRIPTION
**Pull request by nalepae**
**What type of PR is this?**
Other

**What does this PR do? Why is it needed?**
This pull request adds proto for `DataColumnIdentifier`, `DataColumnSidecar`, `DataColumnSidecarsByRangeRequest` and `MetadataV2`.

These are used for peerDAS.

**Other notes for review**
Please read [peerDAS from scratch](https://hackmd.io/@manunalepa/peerDAS) before reviewing this pull request.

**Acknowledgements**
- [x] I have read [CONTRIBUTING.md](https://github.com/prysmaticlabs/prysm/blob/develop/CONTRIBUTING.md).
- [x] I have made an appropriate entry to [CHANGELOG.md](https://github.com/prysmaticlabs/prysm/blob/develop/CHANGELOG.md).
- [x] I have added a description to this PR with sufficient context for reviewers to understand this PR.

### CODE REVIEW

File: proto/prysm/v1alpha1/BUILD.bazel [hunk 1]
```diff
@@ -263,8 +281,10 @@ ssz_gen_marshal(
     objs = [
         "BeaconBlocksByRangeRequest",
         "BlobSidecarsByRangeRequest",
+        "DataColumnSidecarsByRangeRequest",
```
Comment (terencechain) [inline proto/prysm/v1alpha1/BUILD.bazel:284]: `DataColumnSidecarsByRangeRequest` and `MetaDataV2` are not defined anywhere. I think we mistakenly carried them over from the peer das branch

Comment (nalepae) [reply proto/prysm/v1alpha1/BUILD.bazel]: Fixed in 3324b1def2683893a488eea42137b86ebb17efaf.

File: proto/prysm/v1alpha1/BUILD.bazel [hunk 2]
```diff
@@ -174,6 +174,11 @@ ssz_electra_objs = [
     "SignedConsolidation",
 ]
 
+ssz_fulu_objs = [
+    "DataColumnIdentifier",
```
Comment (terencechain) [inline proto/prysm/v1alpha1/BUILD.bazel:178]: Same with `DataColumnIdentifier`, but feel free to merge it. I'm sure you will add `DataColumnIdentifier` in the subsequent PRs

**Review by terencechain — DISMISSED** [review]

File: proto/prysm/v1alpha1/BUILD.bazel (hunk 2 above)
Comment (nalepae) [reply proto/prysm/v1alpha1/BUILD.bazel:178]: Actually I did the opposite in 83ffc531830b29ffc17e31f1521c29c65466dd70: I added everything needed at once.

Comment (nalepae) [reply proto/prysm/v1alpha1/BUILD.bazel:178]: That is to say:
- `DataColumnIdentifier`,
- `DataColumnSidecar`,
- `DataColumnSidecarsByRangeRequest`, and
- `MetadataV2`

**Review by terencechain — APPROVED** [review]

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**